channel/handlers.go

```diff
@@ -762,6 +799,335 @@ func (server Server) playerUseInventoryItem(conn mnet.Client, reader mpacket.Rea
+	}
+
+	_ = server.warpPlayer(plr, dstField, portal)
+}
+
+func (server *Server) playerUseScroll(conn mnet.Client, reader mpacket.Reader) {
+	plr, err := server.players.getFromConn(conn)
+	if err != nil {
+		return
+	}
+
+	scrollSlot := reader.ReadInt16() // USE inv slot (scroll)
+	targetSlot := reader.ReadInt16() // equip slot (negative if equipped)
+
+	if targetSlot < -100 {
+		plr.send(packetPlayerNoChange())
+		return
+	}
+
+	// Load items
+	scroll := findUseItemBySlot(plr, scrollSlot)
+	equip := findEquipBySlot(plr, targetSlot)
+	if scroll == nil || equip == nil || scroll.amount < 1 || equip.amount != 1 {
+		plr.send(packetPlayerNoChange())
+		return
+	}
+
+	// Data and basic checks
+	scrollMeta, err := nx.GetItem(scroll.id)
+	if err != nil {
+		plr.send(packetPlayerNoChange())
+		return
+	}
+	// Type compatibility and basic checks
+	if !validateScrollTarget(scroll.id, equip.id) {
+		plr.send(packetPlayerNoChange())
+		return
+	}
+	if int(scrollMeta.Success) == 0 || equip.getSlots() == 0 {
+		plr.send(packetPlayerNoChange())
+		return
+	}
+
+	// Consume scroll
+	if _, err := plr.takeItem(scroll.id, scrollSlot, 1, 2); err != nil {
+		plr.send(packetPlayerNoChange())
+		return
+	}
+
+	// consume slot
+	equip.setSlots(equip.getSlots() - 1)
+
+	// Roll success/failure
+	rand.Seed(time.Now().UnixNano())
+	successRoll := rand.Intn(100)
+
+	if successRoll < int(scrollMeta.Success) {
+		// Success: apply stats, decrement slot, increment scroll count
+		applyScrollEffects(equip, scrollMeta)
+		incrementScrollCount(equip)
+
+		// Persist and update in-memory slice
+		equip.save(plr.id)
+		plr.updateItem(*equip)
+
+		// Send full item update so client refreshes stats/slots
+		plr.send(packetInventoryAddItem(*equip, true))
+		// Optional: refresh avatar appearance (won't change looks, but safe)
+		plr.send(packetInventoryChangeEquip(*plr))
+		plr.send(packetUseScroll(plr.id, true, false, false))
+	} else {
+		curseRoll := rand.Intn(100)
+		if curseRoll < int(scrollMeta.Cursed) {
+			// Destroy the equip
+			plr.removeItem(*equip)
+			plr.send(packetUseScroll(plr.id, false, true, false))
+		} else {
+			// Normal fail (slot consumed): persist and send full update too
+			equip.save(plr.id)
+			plr.updateItem(*equip)
+			plr.send(packetInventoryAddItem(*equip, true))
+			plr.send(packetUseScroll(plr.id, false, false, false))
+		}
+	}
+}
+
+func getItemType(itemID int32) int32 {
+	return itemID / 10000
+}
+
+func itemTypeToScrollType(itemID int32) int32 {
+	return (getItemType(itemID) % 100) * 100
+}
+
+func getScrollType(itemID int32) int32 {
+	return (itemID % 10000) - (itemID % 100)
+}
+
+// validateScrollTarget performs basic compatibility checks between the scroll and target equip.
+func validateScrollTarget(scrollID int32, equipID int32) bool {
+	return itemTypeToScrollType(equipID) == getScrollType(scrollID)
+}
+
+// applyScrollEffects mutates the equip with the scroll increments from NX.
+func applyScrollEffects(equip *item, scroll nx.Item) {
```

Comment:
This should be changed into an item method to keep in line with the rest of the codebase